### PR TITLE
Script to update a candidate's support and opposed keys in the candidate's json file

### LINKED ISSUE #51: candidate calculation - support
Follow instructions for "support" in the below link https://github.com/opensandiego/sdvv-backend/blob/master/candidate_calc_description.md

### PR DESCRIPTION
**Pull request by robertgz**
The **candidate_calc_outside_spending.js** script reads from the candidate_information Google Sheet and from the Netfile API transaction CSV files. The script updates the candidate JSON files with the outside spending amounts for and against each candidate.

Closes #51, #52

### CODE REVIEW

File: src/scripts/candidate_calc_outside_spending.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+ * @property {string} Office - 
+ * @property {string} Year - 
+ */
+
+/**
+ * Reads from a Google Sheet and returns an array of candidate information objects.
+ * @returns {Promise <Candidates[]>}
+ */
+async function getCandidateInformation() {
+let response, fileData;
```
Comment (nancyheiss) [inline src/scripts/candidate_calc_outside_spending.js:26]: Please add indent.

File: src/scripts/candidate_calc_outside_spending.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+ * @property {number} opposedSum - Decimal number for the sum of all spending in opposition to a candidate
+ */
+
+/**
+ * Reads a list of candidates and a list of transactions. Returns the total spent on supporting or opposing each candidate.
+ * @param {string[]} candidateNames - 
+ * @param {Transactions[]} transactions - 
+ * @returns {CandidateSpending[]}
+ */
+// function getSpendingAmounts(candidateNames, transactions) {
```
Comment (nancyheiss) [inline src/scripts/candidate_calc_outside_spending.js:125]: Looks like this comment can be removed?

File: src/scripts/candidate_calc_outside_spending.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+ * @property {string} candidateName - Name of the Candidate as 'First Last'
+ * @property {string} office - The office the Candidate is running for
+ * @property {string} year - The election year that the Candidate running in
+ * @property {number} supportSum - Decimal number for the sum of all spending in support of a candidate
+ * @property {number} opposedSum - Decimal number for the sum of all spending in opposition to a candidate
+ */
+
+/**
+ * Reads a list of candidates and a list of transactions. Returns the total spent on supporting or opposing each candidate.
+ * @param {string[]} candidateNames -
```
Comment (nancyheiss) [inline src/scripts/candidate_calc_outside_spending.js:121]: candidateNames or candidateInfomation?

File: src/scripts/candidate_calc_outside_spending.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+/**
+ * For each candidate this writes the total outside spending to support and oppose them.  
+ * @param {CandidateSpending[]} outsideSpending 
+ */
+function saveOutsideSpendingToJSON(outsideSpending) {
+  
+  for (const candidate of outsideSpending) {
+
+    // Replace all spaces in candidate names with underscores '_'
+    const candidatePathName = candidate.candidateName.split(' ').join('_').toLowerCase();
```
Comment (nancyheiss) [inline src/scripts/candidate_calc_outside_spending.js:163]: Would it be better to use candidate.replaceAll(' ', '_').toLowerCase();

File: src/scripts/candidate_calc_outside_spending.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+/**
+ * Entry function of script
+ */
+async function calculateOutsideSpending(){
+
+  const candidateInformation = await getCandidateInformation();
+
+  let transactions = await getFilteredTransactions();
```
Comment (nancyheiss) [inline src/scripts/candidate_calc_outside_spending.js:199]: Why is transactions declared with let while the other variables are const?

File: src/scripts/candidate_calc_outside_spending.js (hunk 4 above)
Comment (robertgz) [reply src/scripts/candidate_calc_outside_spending.js:163]: String.prototype.replaceAll() does not appear to be currently supported in Node.js.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#Browser_compatibility

**Pull request merged**